leakpro/attacks/gia_attacks/invertinggradients.py

```diff
@@ -0,0 +1,167 @@
+"""Geiping, Jonas, et al. "Inverting gradients-how easy is it to break privacy in federated learning?."."""
+import torch
+from torch.utils.data import DataLoader, TensorDataset
+
+from leakpro.attacks.gia_attacks.abstract_gia import AbstractGIA
+from leakpro.import_helper import Callable, Self
+from leakpro.metrics.attack_result import GIAResults
+from leakpro.user_inputs.abstract_gia_input_handler import AbstractGIAInputHandler
+
+
+class InvertingGradients(AbstractGIA):
+    """Gradient inversion attack by Geiping et al."""
+
+    def __init__(self: Self, handler: AbstractGIAInputHandler, configs: dict) -> None:
+        super().__init__(handler)
+        self.handler = handler
+        self.logger.info("Inverting gradient initialized :)")
+        self.t_v_scale = configs.get("total_variation")
```

Comment:
Please add a function to ensure these parameters are within correct ranges, see MIA attacks.